Update GitHub lessons

Aariq opened the pull request:
Fairly minor changes

- Narrow scope of .gitignore slide so it doesn't overlap with `git` lesson so much
- Clarify exercise on forking and cloning

I'll need to remember to create two demo repos before these lessons: one that learners all have permission to push to and another public repo that they do not have permission to push to (for fork-clone-PR practice)

Aariq committed:
small updates

Aariq committed:
update gitignore

Aariq committed:
clarify activies, add resources slide

Aariq committed:
add links to slides to syllabus

diazrenata commented on lessons/5-github-basics/slides.qmd:
```diff
@@ -75,15 +59,34 @@ Figure 2 from [@bryan2018]
 -   Push local changes to GitHub
 -   Pull GitHub changes to local
 
+## Ignoring files {.smaller}
```
I think this will be a good complement to introducing .gitignore in the git lesson :)

Aariq committed:
merge main + render

Merge commit '3eec0b52ae8727ead0534a5f1759cf7960571306'

#Conflicts:
#	_freeze/index/execute-results/html.json
#	index.qmd
#	lessons/5-github-basics/slides.qmd
#	lessons/6-github-collab/slides.qmd
#	schedule.csv
